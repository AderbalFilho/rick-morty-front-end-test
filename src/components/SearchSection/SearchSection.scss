.search-section {
  align-items: center;
  display: flex;
  flex-direction: column;
}

.search-section__logo {
  margin: 19px 40px 47px;
  max-width: 502px;
  width: calc(100% - 80px);

  @media (min-width: 768px) {
    margin: 38px 0 74px;
    width: 100%;
  }
}

.search-section__input-container {
  display: flex;
}

.search-section__button {
  margin-left: 8px;

  @media (min-width: 768px) {
    margin-left: 16px;
  }
}
