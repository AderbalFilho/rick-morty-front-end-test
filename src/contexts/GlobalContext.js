import React, { createContext, useState } from 'react';

export const GlobalContext = createContext();

export const GlobalStorage = ({ children }) => {
  const [searchTerm, setSearchTerm] = useState('');
  const [characters, setCharacters] = useState([]);
  const [isLoading /*, setIsLoading*/] = useState(false);

  function handleSearch() {
    // Do search
    setCharacters([
      /* result */
    ]);
  }

  return (
    <GlobalContext.Provider
      value={{ searchTerm, setSearchTerm, characters, handleSearch, isLoading }}
    >
      {children}
    </GlobalContext.Provider>
  );
};
