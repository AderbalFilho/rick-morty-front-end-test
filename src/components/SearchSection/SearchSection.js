import React, { useContext } from 'react';

import { GButton, GInput } from '@ui';
import { GlobalContext } from '@contexts/GlobalContext';

import logo from '@assets/imgs/logo.png';

import './SearchSection.scss';

function SearchSection() {
  const { searchTerm, setSearchTerm, handleSearch } = useContext(GlobalContext);

  return (
    <section className="search-section">
      <img
        src={logo}
        alt="Rick and Morty logo"
        className="search-section__logo"
      />
      <div className="search-section__input-container">
        <GInput
          type="text"
          placeholder="Search characters"
          value={searchTerm}
          change={({ target }) => setSearchTerm(target.value)}
        />
        <GButton classes="search-section__button" onClick={handleSearch}>
          Search
        </GButton>
      </div>
    </section>
  );
}

export default SearchSection;
