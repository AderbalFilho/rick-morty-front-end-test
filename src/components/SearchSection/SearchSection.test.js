import React from 'react';
import TestRenderer from 'react-test-renderer';

import SearchSection from './SearchSection';
import { GlobalStorage } from '@contexts/GlobalContext';

describe('SearchSection', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(
      <GlobalStorage>
        <SearchSection />
      </GlobalStorage>
    );
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });
});
