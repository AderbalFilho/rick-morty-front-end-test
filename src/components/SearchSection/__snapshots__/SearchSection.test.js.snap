// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSection matches snapshot 1`] = `
<section
  className="search-section"
>
  <img
    alt="Rick and Morty logo"
    className="search-section__logo"
    src=""
  />
  <div
    className="search-section__input-container"
  >
    <input
      className="g-input"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Search characters"
      type="text"
      value=""
    />
    <button
      className="g-button search-section__button"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</section>
`;
